src/analysis/cauchy_equation.lean

```diff
@@ -0,0 +1,156 @@
+/-
+Copyright (c) 2022 Mantas Bakšys. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mantas Bakšys
+-/
+import analysis.normed_space.pointwise
+import measure_theory.measure.haar_lebesgue
+
+/-!
+# Cauchy's Functional Equation
+
+This file contains the classical results about the Cauchy's functional equation
+`f (x + y) = f x + f y` for functions `f : ℝ → ℝ`. In this file, we prove that the solutions to this
+equation are linear up to the case when `f` is a Lebesgue measurable functions, while also deducing
+intermediate well-known variants.
+-/
+
+open add_monoid_hom measure_theory measure_theory.measure metric nnreal set
+open_locale pointwise topological_space
+
+variables {ι : Type*} [fintype ι] {s : set ℝ} {a : ℝ}
+
+local notation `ℝⁿ` := ι → ℝ
+
+/-- **Cauchy's functional equation**. An additive monoid homomorphism automatically preserves `ℚ`.
+See also `add_monoid_hom.to_rat_linear_map`. -/
+lemma add_monoid_hom.is_linear_map_rat (f : ℝ →+ ℝ) : is_linear_map ℚ f :=
+⟨map_add f, map_rat_cast_smul f ℝ ℝ⟩
+
+lemma measure_theory.measure.ne_zero {X : Type*} [nonempty X] [topological_space X]
+  {m : measurable_space X} (μ : measure X) [μ.is_open_pos_measure] : μ ≠ 0 :=
+λ h, is_open_univ.measure_ne_zero μ univ_nonempty $ by rw [h, coe_zero, pi.zero_apply]
+
+lemma exists_zero_nhds_bounded (f : ℝ →+ ℝ) (h : measurable f) :
+  ∃ s, s ∈ 𝓝 (0 : ℝ) ∧ bounded (f '' s) :=
+begin
+  obtain ⟨r, hr⟩ := exists_pos_preimage_ball f (0 : ℝ) volume.ne_zero,
+  refine ⟨_, sub_mem_nhds_zero_of_add_haar_pos volume (f ⁻¹' ball 0 r) (h $ measurable_set_ball) hr,
+    _⟩,
+  rw image_sub,
+  exact (bounded_ball.mono $ image_preimage_subset _ _).sub
+    (bounded_ball.mono $ image_preimage_subset _ _),
+end
+
+lemma additive_continuous_at_zero_of_bounded_nhds_zero (f : ℝ →+ ℝ) (hs : s ∈ 𝓝 (0 : ℝ))
+  (hbounded : bounded (f '' s)) : continuous_at f 0 :=
```

Comment:
If you upgrade `f` to a `ℚ`-linear map, then you can use `linear_map.bound_of_shell` or `linear_map.bound_of_ball_bound`. Again, you can upgrade your lemma to a real TVS.